File: Practice/33/Python/33.py

def create(arr_size, arr_start = 0, arr_step = 0):
    return [i*arr_step + arr_start for i in range(arr_size)]

# метод сортировки вставками
def sort (arr):
    i = 1
    while i < len(arr):
        j=i
        while j > 0 and arr[j-1] > arr[j]:
            chg = arr[j - 1]
            arr[j - 1] = arr[j]
            arr[j] = chg
            j-= 1
        i+= 1
    return arr

_print = print
def print(arr):
    _print(arr)
    return arr

arr_size  = int(input())
arr_start = int(input())
arr_step  = int(input())
arr = create(arr_size, arr_start, arr_step)
arr = sort(arr)
print(arr)

File: Practice/33/С++/33/33/33.cpp
#include <iostream>
using namespace std;

int* create(int arr_size, int arr_start = 0, int arr_step = 0) {

    int* arr_new = new int[arr_size];

    for (int i = 0; i < arr_size; i++) {
        arr_new[i] = arr_start;
        arr_start += arr_step;
    }
    return arr_new;
}

int* sort(int* arr, int arr_size) {
    int i = 1;
    while (i < arr_size) {
        int j = i;
        while (j > 0 and arr[j - 1] > arr[j]) {
            swap(arr[j - 1], arr[j]);
            j -= 1;
        }
        i += 1;
    }
    return arr;
}

int* print(int* arr, int arr_size) {
    cout << "[";
    for (int i = 0; i < arr_size; i++) {
        cout << arr[i];
        if (i < arr_size - 1) {
            cout << ", ";
        }
    }
    cout << "]";
    return arr;
}

int main()
{
    int arr_size, arr_start, arr_step;

    cin >> arr_size;
    cin >> arr_start;
    cin >> arr_step;

    int* arr = create(arr_size, arr_start, arr_step);
    print(sort(arr, arr_size), arr_size);

    delete[] arr;
    return 0;
}